bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java

```diff
@@ -171,7 +166,7 @@ public Thing approve(ThingUID thingUID, String label) {
     public synchronized boolean add(DiscoveryResult result) throws IllegalStateException {
         if (result != null) {
             ThingUID thingUID = result.getThingUID();
-            Thing thing = this.thingRegistry.get(thingUID);
+            Thing thing = getExistingThingForDiscoveryResult(result);
```

Comment:
Yes, this sounds like a reasonable approach. Will change it. But thi feature was already there, you might have overlooked it in the first place.